pkg/controller/worker/machines.go

```diff
@@ -227,3 +231,28 @@ func (w *workerDelegate) getServerLabelsForMachine(machineType string, workerCon
 	}
 	return combinedLabels, nil
 }
+
+func (w *workerDelegate) getMergedIgnitionConfig(ctx context.Context, workerConfig *metalv1alpha1.WorkerConfig) (string, error) {
+	var mergedIgnition string
+
+	if workerConfig.ExtraIgnition.Raw != "" {
+		mergedIgnition = workerConfig.ExtraIgnition.Raw
+	}
+
+	if workerConfig.ExtraIgnition.IgnitionSecret != "" {
+		secret := &corev1.Secret{}
+		secretKey := client.ObjectKey{Namespace: w.worker.Namespace, Name: workerConfig.ExtraIgnition.IgnitionSecret}
+		if err := w.client.Get(ctx, secretKey, secret); err != nil {
+			return "", fmt.Errorf("failed to get ignition secret %s: %w", workerConfig.ExtraIgnition.IgnitionSecret, err)
+		}
+
+		secretContent, ok := secret.Data["ignition"]
+		if !ok {
+			return "", fmt.Errorf("ignition key not found in secret %s", workerConfig.ExtraIgnition.IgnitionSecret)
+		}
+
+		mergedIgnition += string(secretContent)
```

Comment:
Is concatenation equivalent to merging?

Reply:
I would say it should be fine in this case to concat, since:
- we work with 2 strings
- we cannot cover the error case where same thing is defined in both places. This is up to the user.
- `extraIgnition` should be used for debug/test to put additional ignition on top of what is provided in Secret

What do you think?

Reply:
Looks better with `mergo`.